[docs] Voyager docs for v1.3 (#17505)
yugabyte/yugabyte-db: docs/content/preview/migrate/migrate-steps.md

The options passed to the command are similar to the [`yb-voyager export schema`](#export-schema) command. To export only a subset of the tables, pass a comma-separated list of table names in the `--table-list` argument.

#### Accelerate data export (Optional for MySQL/Oracle only)

If you want to speed up the data export for MySQL/Oracle, you can set the environment variable `EXPERIMENTAL_FAST_EXPORT=1` when you run `export data`.

Consider the following caveats before using the feature:

- You need to perform additional steps when you [prepare the source database](#prepare-the-source-database).
- Some data types are unsupported. For a detailed list, refer to [datatype mappings](../reference/datatype-mapping-mysql/).

> - [--parallel-jobs](../reference/yb-voyager-cli/#parallel-jobs) argument is unsupported.

Reviewer: Can you please change it to `parallel-jobs argument will have no effect`

The options passed to the command are similar to the [`yb-voyager export schema`](#export-schema) command. To export only a subset of the tables, pass a comma-separated list of table names in the `--table-list` argument.

> #### Accelerate data export (Optional for MySQL/Oracle only)

Reviewer 2: ```suggestion
#### Accelerate data export
```

> For MySQL or Oracle, you can optionally speed up data export by setting the environment variable `EXPERIMENTAL_FAST_EXPORT=1` when you run `export data` using yb-voyager.

Reviewer 3: Shouldn't it be `For MySQL and Oracle` instead of `For MySQL or Oracle`

> For MySQL and Oracle, you can optionally speed up data export by setting the environment variable `EXPERIMENTAL_FAST_EXPORT=1` when you run `export data` using yb-voyager.

Reviewer: @aishwarya24 can you please replace all instances of `EXPERIMENTAL_FAST_EXPORT` to `BETA_FAST_DATA_EXPORT` ? We discussed internally decided that this was more inviting for users to try out.

> If you want to speed up data export for MySQL/Oracle, you can set the environment variable `EXPERIMENTAL_FAST_EXPORT=1` when you run `export data` using yb-voyager.

Reviewer 2: ```suggestion
For MySQL or Oracle, you can optionally speed up data export by setting the environment variable `EXPERIMENTAL_FAST_EXPORT=1` when you run `export data` using yb-voyager.
```

Note that there are some special cases involving sequences such as the following:

> - In MySQL, auto-increment column is migrated as a normal column with a sequence attached to it.

Reviewer: `For MySQL, ...`

- Some data types are unsupported. For a detailed list, refer to [datatype mappings](../reference/datatype-mapping-mysql/).
- [--parallel-jobs](../reference/yb-voyager-cli/#parallel-jobs) argument will have no effect.
- SSL is unsupported.
- In MySQL RDS, writes are not allowed during the data export process.

> {{< note title="Sequence migration considerations" >}}

Reviewer 2: what's the link? if EXPERIMENTAL_FAST_EXPORT is using sequence migration, that should be worked into/explained in the opening paragraph of this section, rather than being brought up in an otherwise unrelated-looking note at the end of the section

Reviewer: Agreed, it is unrelated. @aishwarya24 can you pls move this to a separate section similar to `Accelerate data export for MySQL or Oracle`

Reviewer: @aishwarya24 I believe this is still a note. Can you please convert it to a separate section as mentioned? Thanks

> - In PostgreSQL, `SERIAL` datatype and `GENERATED AS IDENTITY` columns use sequence object internally, so resume values for them are also generated during data export.

Reviewer: `For PostgreSQL, ...`

> - For Oracle where sequences are not attached to a column, resume value generation is unsupported with EXPERIMENTAL_FAST_EXPORT=1 environment variable.

Reviewer: Can you pls modify it to `For Oracle where sequences are not attached to a column, resume value generation is unsupported` 

`with EXPERIMENTAL_FAST_EXPORT=1 environment variable` is redundant given that it is under that section.

Sequence migration consists of two steps: sequence creation and setting resume value (resume value refers to the `NEXTVAL` of a sequence on a source database). A sequence object is generated during export schema and the resume values for sequences are generated during export data. These resume values are then set on the target database just after the data is imported for all tables.

Note that there are some special cases involving sequences such as the following:

- In MySQL, auto-increment column is migrated as a normal column with a sequence attached to it.
- In PostgreSQL, `SERIAL` datatype and `GENERATED AS IDENTITY` columns use sequence object internally, so resume values for them are also generated during data export.

> - For PostgreSQL and Oracle where sequences are not attached to a column, resume value generation is unsupported with EXPERIMENTAL_FAST_EXPORT=1 environment variable.

Reviewer: @aishwarya24 can you pls additionally add this to the list of caveats under the section `Accelerate data export for MySQL or Oracle` ?

        –-data-dir </path/to/files/dir/> \
        --file-table-map <filename1:table1,filename2:table2> \
        --delimiter <DELIMITER> \
        –-has-header \

>         --null-string

Reviewer: `--null-string "<NULL_STRING>"`
